plugins/GSdx/Renderers/Common/GSRenderer.cpp

```diff
@@ -75,6 +75,12 @@ bool GSRenderer::CreateDevice(GSDevice* dev)
 	m_dev = dev;
 	m_dev->SetVSync(m_vsync);
 
+	extern int g_aspect;
```

Comment:
Do we need this extern here? Can we just call SetAspectRatio and SetZoom where this is called.

Reply:
SetAspectZoom gets called before a renderer is ready, so it gets stored in the globals instead (GS.cpp). On render creation (and m_dev is set), it uses the stored global for initial state.

Reply:
Right, but create device is also called in gs.cpp during init so I'm confused why you can't just use this pattern.

```cpp
if(!s_gs->CreateDevice(dev))
{
	GSclose();
	return -1;
}

s_gs->SetZoom(g_zoom);
s_gs->SetAspectRatio(g_aspect);
```

Reply:
Yes, it can be moved there and the globals made static again. I'll give it a try.